### PR TITLE
[SPARK-26995][K8S] Make ld-linux-x86-64.so.2 visible to snappy native library under /lib in docker image with Alpine Linux

### PR DESCRIPTION
**Pull request by LucaCanali**
Running Spark in Docker image with Alpine Linux 3.9.0 throws errors when using snappy. 

The issue can be reproduced for example as follows: `Seq(1,2).toDF("id").write.format("parquet").save("DELETEME1")` 
The key part of the error stack is as follows `SparkException: Task failed while writing rows. .... Caused by: java.lang.UnsatisfiedLinkError: /tmp/snappy-1.1.7-2b4872f1-7c41-4b84-bda1-dbcb8dd0ce4c-libsnappyjava.so: Error loading shared library ld-linux-x86-64.so.2: Noded by /tmp/snappy-1.1.7-2b4872f1-7c41-4b84-bda1-dbcb8dd0ce4c-libsnappyjava.so)` 

The source of the error appears to be that libsnappyjava.so needs ld-linux-x86-64.so.2 and looks for it in /lib, while in Alpine Linux 3.9.0 with libc6-compat version 1.1.20-r3 ld-linux-x86-64.so.2 is located in /lib64.
Note: this issue is not present with Alpine Linux 3.8 and libc6-compat version 1.1.19-r10 

## What changes were proposed in this pull request?

A possible workaround proposed with this PR is to modify the Dockerfile by adding a symbolic link between /lib and /lib64 so that linux-x86-64.so.2 can be found in /lib. This is probably not the cleanest solution, but I have observed that this is what happened/happens already when using Alpine Linux 3.8.1 (a version of Alpine Linux which was not affected by the issue reported here).

## How was this patch tested?

Manually tested by running a simple workload with spark-shell, using docker on a client machine and using Spark on a Kubernetes cluster. The test workload is: `Seq(1,2).toDF("id").write.format("parquet").save("DELETEME1")` 

Added a test to the KubernetesSuite / BasicTestsSuite

### CODE REVIEW

**Comment by vanzin** [comment]
Any easy way to change an existing integration test to use snappy?

**Comment by vanzin** [comment]
(And, BTW, the PR title should explain the fix, not the problem.)

**Comment by LucaCanali** [comment]
Thanks @vanzin for the comments. I have updated the PR title. I'll look at the test for this.

**Comment by vanzin** [comment]
Sorry, but the PR title still does not explain the fix.

e.g. "Make Alpine lib directory visible to snappy native library in docker image."

Or something.

**Comment by vanzin** [comment]
ok to test

**Review by felixcheung — COMMENTED** [review]
looks reasonable...

File: examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLReadParquetExample.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+package org.apache.spark.examples.sql
+
+import org.apache.spark.sql.SparkSession
+
+// Example/test code: reads a Parquet file from the provided Spark examples
+// Optional argument: path to dir with examples (typically SPARK_HOME)
+// This is also used by the Kubernetes Tests Suite
+object SparkSQLReadParquetExample {
```
Comment (vanzin) [inline examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLReadParquetExample.scala]: Can you use the existing `SparkSQLExample` for this? Seems overkill to add a new example just so you can run it in the k8s tests.

I was hoping there was something simple like setting `spark.io.compression.codec=snappy`, but I'm not sure SparkPi would actually hit that. (Maybe `SparkRemoteFileTest`, which is run already, does, though?)

If the above doesn't work, or `SparkSQLExample` is too slow or doesn't work for this, then I think in this case it's better to skip the test until we can separate examples from k8s integration tests.

Comment (LucaCanali) [reply examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLReadParquetExample.scala]: Good point, thanks. Setting `spark.io.compression.codec=snappy` seems to work to test this.

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala [hunk 2]
```diff
@@ -81,6 +81,12 @@ private[spark] trait BasicTestsSuite { k8sSuite: KubernetesSuite =>
       })
   }
 
+  test("Run SparkPi with spark.io.compression.codec=snappy for SPARK-26995.", k8sTestTag) {
```
Comment (vanzin) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala]: Sorry to be a pain about these things, but could you instead change an existing test?

Integration tests take much longer to run than unit tests, the more we can test with the same invocation, the better.

Comment (LucaCanali) [reply resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala]: Sure, no problem.

**Comment by vanzin** [comment]
Merging to master.

**Comment by dongjoon-hyun** [comment]
Hi, All. I'll backport this to `branch-2.4`.

**Pull request closed without merging**